### PR TITLE
feat(cli): add private-api template

### PR DESCRIPTION
**Pull request by hasanaburayyan**
Leaving in draft until: https://github.com/winglang/wing/pull/5659 merges

## Checklist

- [ ] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Review by Chriscbr — APPROVED** [review]

File: apps/wing/project-templates/wing/private-api/wing.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[ tf-aws ]
+vpc = "new"
+vpc_lambda = true
```
Comment (eladb) [inline apps/wing/project-templates/wing/private-api/wing.toml:3]: Add some comments here so that users will have a sense of the options.

File: apps/wing/project-templates/wing/private-api/main.w [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+let api = new cloud.Api();
+api.get("/", inflight () => {
+  return {
+    status: 200,
+    body: "hello, from within the VPC!"
+  };
+});
+
+
+let url = api.url;
```
Comment (eladb) [inline apps/wing/project-templates/wing/private-api/main.w:13]: why is this needed?

Comment (hasanaburayyan) [reply apps/wing/project-templates/wing/private-api/main.w:13]: Ill check but I dont think its needed, I copy and pasted this template from somewhere, Ill remove
